driver-core/src/main/com/mongodb/spi/dns/InetAddressResolver.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spi.dns;
+
+import com.mongodb.MongoClientSettings;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+
+/**
+ * This interface defines operations for looking up host names.
+ *
+ * <p> The <i>system-wide resolver</i> can be customized by
+ * <a href="InetAddressResolverProvider.html#system-wide-resolver">
+ * deploying an implementation</a> of {@link InetAddressResolverProvider}.
+ * <p>
+ * @since 4.10
+ */
+public interface InetAddressResolver {
+    /**
+     * Given the name of a host, returns a list of IP addresses of the requested
+     * address family associated with a provided hostname.
+     *
+     * <p> {@code host} should be a machine name, such as "{@code www.example.com}",
+     * not a textual representation of its IP address. No validation is performed on
+     * the given {@code host} name: if a textual representation is supplied, the name
+     * resolution is likely to fail and {@link UnknownHostException} may be thrown.
```

Comment:
[optional]
Am I the only one to whom this text adopted from the Java SE docs appears to be weird? This interface is supposed to document requirements for a user who implements the interface. Whether to perform a validation and whether to fail if the argument is a string representation of an IP address, is up to the user. The documentation may say that the implementation does not have to support string representations of IP addresses and does not have to do any validation. But instead, the documentation is formulated as if it's a documentation of a provided implementation, which it is not.

Reply:
It's better to make the requirements of our `InetAddressResolver` as similar as possible to those of the Java SE API `InetAddressResolver`. I.e., we also should not require `InetAddressResolver` to resolve textual representations of IP addresses.

Also, I think we should not copy the text from the Java SE API for this interface: instead of expressing it as a requirement for the implementor (our user), it is expressed as it if it written for the consumer (the driver maintainers), which is wrong.

Reply:
Done.  I added a check in `ServerAddressWithResolver` for IP literals.